### PR TITLE
Fix handling of JMail error conditions

### PR DESCRIPTION
**Pull request by mbabker**
#### Summary of Changes

In an error condition `JMail::Send()` can return either a boolean false (if the site's mail function is turned off in the global config) or the result of `JError::raiseNotice()` (which is a `JException` object).  This updates core to explicitly check for a boolean true return value (which is the only return on a successful send) instead of implicit checks for a boolean false or `Exception` object and also corrects an incorrect method reference (`Exception` does not have a `getError()` method).
#### Testing Instructions

Mainly review, otherwise validate that mail correctly sends in the changed methods AND error conditions are handled correctly.

### CODE REVIEW

File: installation/controller/install/email.php [hunk 1]
```diff
@@ -132,7 +132,10 @@ public function execute()
 
 		try
 		{
-			$mail->Send();
+			if ($mail->Send() !== true)
+			{
+				$app->enqueueMessage(JText::_('INSTL_EMAIL_NOT_SENT'), 'notice');
```
Comment (andrepereiradasilva) [inline installation/controller/install/email.php]: If the mail is not sent some problem existed. So, shouldn't be a warning, instead of a notice?

Comment (mbabker) [inline installation/controller/install/email.php]: It's a copy paste of the error handling in the caught exception.  I wasn't intending to get into dealing with message types.  And actually it should be an error, not a warning or notice (note I personally use the [PSR-3 Interface](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-3-logger-interface.md) as a guide for deciding message types for both logging and UI display).

Comment (andrepereiradasilva) [inline installation/controller/install/email.php]: ok then an error

Comment (andrepereiradasilva) [inline installation/controller/install/email.php]: so, can you change this to a error message type as talked above?

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 8828440db1d03012ad68ffcaf7941aea1df6ceee<br /><br />on code review. thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9778">issues.joomla.org/joomla-cms/9778</a>.</sub>

**Comment by yvesh** [comment]
I have tested this item :white_check_mark: successfully on 8828440db1d03012ad68ffcaf7941aea1df6ceee<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9778">issues.joomla.org/joomla-cms/9778</a>.</sub>

**Comment by mbabker** [comment]
This has fallen out of sync and I'm not interested in pursuing this anymore.

**Pull request closed without merging**